File: client/web/sudoo/lib/domain/model/discovery/product.dart
import 'package:sudoo/domain/model/discovery/file.dart';

import 'category.dart';
import 'extras.dart';
import 'supplier_info.dart';

class Product {
  final String productId;
  final String sku;
  final String name;
  final String brand;
  final String description;
  double price;
  double listedPrice;
  final int amount;
  final int soldAmount;
  final double rate;
  int discount;
  final DateTime? startDateDiscount;
  DateTime? endDateDiscount;
  bool saleable;
  final int weight;
  final int height;
  final int width;
  final int length;
  final List<File> images;
  final SupplierInfo supplier;
  final List<Category> categories;
  final Extras? extras;

  Product(
    this.productId,
    this.sku,
    this.name,
    this.brand,
    this.description,
    this.price,
    this.listedPrice,
    this.amount,
    this.soldAmount,
    this.rate,
    this.discount,
    this.startDateDiscount,
    this.endDateDiscount,
    this.saleable,
    this.weight,
    this.height,
    this.width,
    this.length,
    this.images,
    this.supplier,
    this.categories,
    this.extras,
  );
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/entity/discovery/Product.kt
package com.sudo248.sudoo.domain.entity.discovery

import java.time.LocalDateTime


/**
 * **Created by**
 *
 * @author *Sudo248*
 * @since 23:09 - 12/03/2023
 */
data class Product(
    val productId: String,
    val supplierId: String ,
    val name: String,
    val description: String,
    val brand: String ,
    val sku: String,
    val images: List<String>,
    val price: Double,
    val listedPrice: Double,
    val amount: Int,
    val soldAmount: Int,
    val rate: Float,
    val discount: Int,
    val startDateDiscount: LocalDateTime,
    val endDateDiscount: LocalDateTime,
    val saleable: Boolean,
    val weight: Int,
    val height: Int,
    val length: Int,
    val width: Int,
    val extras: ProductExtras,
    val supplier: SupplierInfo? = null,
    val categories: List<CategoryInfo>? = null,
) : java.io.Serializable
